rest-new/src/main/java/org/jboss/pnc/rest/endpoints/BuildEndpointImpl.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014-2019 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.rest.endpoints;
+
+import org.jboss.pnc.dto.Artifact;
+
+import java.lang.invoke.MethodHandles;
+
+import javax.inject.Inject;
+
+import org.jboss.pnc.dto.Build;
+import org.jboss.pnc.dto.BuildConfigurationRevision;
+import org.jboss.pnc.dto.BuildPushResult;
+import org.jboss.pnc.dto.BuildRef;
+import org.jboss.pnc.dto.requests.BuildPushRequest;
+import org.jboss.pnc.dto.response.Graph;
+import org.jboss.pnc.dto.response.Page;
+import org.jboss.pnc.dto.response.SSHCredentials;
+import org.jboss.pnc.facade.providers.api.BuildPageInfo;
+import org.jboss.pnc.facade.providers.api.BuildProvider;
+import org.jboss.pnc.rest.api.endpoints.BuildEndpoint;
+import org.jboss.pnc.rest.api.parameters.BuildAttributeParameters;
+import org.jboss.pnc.rest.api.parameters.BuildsFilterParameters;
+import org.jboss.pnc.rest.api.parameters.PageParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Honza Brázdil &lt;jbrazdil@redhat.com&gt;
+ */
+public class BuildEndpointImpl extends AbstractEndpoint<Build, BuildRef> implements BuildEndpoint {
```

Comment:
Missing EJB annotation!

Reply:
Yeah, missing annotation but why EJB and not CDI?

Also from [Resteasy doc](https://docs.jboss.org/resteasy/docs/3.6.1.Final/userguide/html/CDI.html):
> If a JAX-RS root resource does not define a scope explicitly, it is bound to the Request scope.

Reply:
Sure go with CDI then!

Reply:
We'll have to sync all the other EndpointImpl to use the same CDI annotation/EJB though

Reply:
Yeah, I have it planned